lib/xnvme_be_spdk_dev.c

```diff
@@ -528,6 +531,9 @@ enumerate_probe_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 		ctrlr_opts->header_digest = 1;
 		ctrlr_opts->data_digest = 1;
 		ctrlr_opts->keep_alive_timeout_ms = 0;
+		if (opts->hostnqn) {
+			strcpy(ctrlr_opts->hostnqn, opts->hostnqn);
```

Comment:
Hmm, perhaps change this to:
``strncpy(ctrlr_opts->hostnqn, opts->hostnqn, SPDK_NVMF_NQN_MAX_LEN)``